release/release_tests.yaml

```diff
@@ -5018,9 +5018,7 @@
   team: core
   cluster:
     byod:
-      runtime_env:
-        - RAY_MAX_LIMIT_FROM_API_SERVER=1000000000
-        - RAY_MAX_LIMIT_FROM_DATA_SOURCE=1000000000
+      post_build_script: byod_increase_ray_max_limit_test.sh
```

Comment:
if it is adding two env variables, why not add a env list here under byod, and add `ENV RAY_MAX_LIMIT_FROM_API_SERVER=1000000000` into the Dockerfile that you generated? that way, you do not need to alter the `.bashrc` file, and it will be much more stable.

we cannot really build on the assumption that `.bashrc` will be always executed for a ray job.

Reply:
Unless we dynamically generate Dockerfile at runtime which is way too complicated. 

We own the infrastructure of bootstrapping ray from anyscale so we should be able to make some assumptions about what file to load at runtime no?

Reply:
I wonder how anyscale builds image with a dynamic list of env today, like this one: https://console.anyscale-staging.com/o/anyscale-internal/configurations/app-config-details/bld_c41b4hfb5h1rqjtbtkxtamkqjp

Reply:
> how anyscale builds image with a dynamic list of env

the Dockerfile it uses is dynamically generated. you can do the same thing. like when env is not empty, insert several lines of `ENV KEY=value` into the Dockerfile.

Reply:
> dynamically generate Dockerfile at runtime which is way too complicated.

why is that complicated?

Reply:
> own the infrastructure of bootstrapping ray from anyscale so we should be able to make some assumptions about what file to load at runtime no?

we have the ownership of the infra, but that does not mean that it is justified to use `bashrc` as an init interface. we still need to design and maintain the interface to be clean and tidy.

Reply:
I see. Sound good. It feel like though that anyscale should publish its api for building its docker image with an arbitrary base image. Seem like I reinvent the wheel for its cluster_env build process.

Reply:
> > dynamically generate Dockerfile at runtime which is way too complicated.
> 
> why is that complicated?

like you are already dynamic sending in this `post_build_script` thing.

what you need to do is create a temp tarball with the dynamically generated Dockerfile (and the post build script) in it, and send the tarball in either via stdin.

Reply:
cc @yifeif too.

Reply:
Also the runtime_env here is probably incorrect or confusing. It doesn't not run as part of the docker entrypoint env, it runs as part of ray.init https://docs.anyscale.com/reference/python-sdk/models#createproductionjobconfig


Should it be invoked through docker run -env instead?

Reply:
Since this discussion is more about product than open source use cases, opening thread in product instead.